pkg/tools/etcd_helper.go

```diff
@@ -484,24 +484,14 @@ func (h *EtcdHelper) SetObj(key string, obj, out runtime.Object, ttl uint64) err
 	return err
 }
 
-// ResponseMeta contains information about the etcd metadata that is associated with
-// an object. It abstracts the actual underlying objects to prevent coupling with etcd
-// and to improve testability.
-type ResponseMeta struct {
-	// TTL is the time to live of the node that contained the returned object. It may be
-	// zero or negative in some cases (objects may be expired after the requested
-	// expiration time due to server lag).
-	TTL int64
-}
-
 // Pass an EtcdUpdateFunc to EtcdHelper.GuaranteedUpdate to make an etcd update that is guaranteed to succeed.
 // See the comment for GuaranteedUpdate for more detail.
-type EtcdUpdateFunc func(input runtime.Object, res ResponseMeta) (output runtime.Object, ttl *uint64, err error)
+type EtcdUpdateFunc func(input runtime.Object, node *etcd.Node) (output runtime.Object, ttl *uint64, err error)
```

Comment:
Yeah the other way around.  I fought but then conceded because he was right - pass what you need, not what you think you'll want.  :)

----- Original Message -----
> >  // Pass an EtcdUpdateFunc to EtcdHelper.GuaranteedUpdate to make an etcd
> >  update that is guaranteed to succeed.
> >  // See the comment for GuaranteedUpdate for more detail.
> > -type EtcdUpdateFunc func(input runtime.Object, res ResponseMeta) (output
> > runtime.Object, ttl *uint64, err error)
> > +type EtcdUpdateFunc func(input runtime.Object, node *etcd.Node) (output
> > runtime.Object, ttl *uint64, err error)
> 
> Yes. I can add node.Expiration and node.ModifiedIndex to ResponseMeta and
> then change e.Helper.Versioner.UpdateObject() to take modified index and
> expiration instead of the whole node, if this is a concern. Based on that
> discussion, I thought @smarterclayton had convinced @lavalamp that its fine
> to pass node, but looks like thats not the case.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/10074/files#r32851841